### PR TITLE
Tests for registration component added

### LINKED ISSUE #135: Home component for login/register
Look at the Figma mockups.

We want the client app to be as minimized as possible. Create a home component where the user can Login or Register. Make use of NgIf in order to only show one form at a time.  

[NgForms](https://angular.io/api/forms/NgForm) are just enough.

### PR DESCRIPTION
**Pull request by PilgrimViis**
I've added registration component logic and added a bit of testing.

### CODE REVIEW

File: Client/src/app/components/auth/registration/registration.component.html [hunk 1]
```diff
@@ -91,7 +94,7 @@ <h1 class="offset-md-6 offset-3">Register</h1>
       Register
     </button>
     <span class="col-md-3">
-      Have already an account?&nbsp;<a href="/login" class="underlineLink">Go to login</a>      
+      Have already an account?&nbsp;<a href="/login" class="underlineLink">Go to login</a>
```
Comment (HellHunterMax) [inline Client/src/app/components/auth/registration/registration.component.html:97]: This should read: Already have an acount?

**Review by HellHunterMax — COMMENTED** [review]
Maybe some kind of Message like Account created Successfully ?
And i just saw the link back to the jokes does not work anymore.
@chrisK00 should look at this. but i just checked if it works and it does ;)

good job imo its really easy to follow even for someone who does not know angular ;)

File: Client/src/app/components/auth/registration/registration.component.html (hunk 1 above)
Comment (PilgrimViis) [reply Client/src/app/components/auth/registration/registration.component.html:97]: > And i just saw the link back to the jokes does not work anymore.

What link? /jokes?

Comment (HellHunterMax) [reply Client/src/app/components/auth/registration/registration.component.html:97]: Sorry I should have been more clear. The left  button on the navbar does not link back to jokes.

File: Server/JokesOnYou.Web.Api/Models/Request/UserRegisterDTO.cs [hunk 2]
```diff
@@ -8,9 +9,13 @@ public class UserRegisterDto
         [EmailAddress]
         public string Email { get; set; }
         [Required]
+        [MinLength(4)]
         public string UserName { get; set; }
         [Required]
         public string Password { get; set; }
-
+        [Required]
```
Comment (chrisK00) [inline Server/JokesOnYou.Web.Api/Models/Request/UserRegisterDTO.cs]: Doesnt have to be required, can be false by default. Should we also check that the user is above a specific age?

Comment (Niisc) [reply Server/JokesOnYou.Web.Api/Models/Request/UserRegisterDTO.cs]: We probably should, i dont think its legal to show nsfw stuff to people under 18 or something

Comment (HellHunterMax) [reply Server/JokesOnYou.Web.Api/Models/Request/UserRegisterDTO.cs]: nsfw does not mean Sexy stuff per se and we wont even allow like naked stuff. (not that we have pictures yet)
Should just be there for like Swear words and stuff? so no date required for that.

File: Client/src/app/components/auth/registration/registration.component.html (hunk 1 above)
Comment (PilgrimViis) [reply Client/src/app/components/auth/registration/registration.component.html:97]: > 
> 
> Sorry I should have been more clear. The left button on the navbar does not link back to jokes.

No problem. Done.

File: Server/JokesOnYou.Web.Api/Models/Request/UserRegisterDTO.cs (hunk 2 above)
Comment (PilgrimViis) [reply Server/JokesOnYou.Web.Api/Models/Request/UserRegisterDTO.cs]: Done

**Review by chrisK00 — APPROVED** [review]

**Pull request merged**